src/global_mapper/global_mapper_params.h

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017 Massachusetts Institute of Technology
+#pragma once
+
+namespace global_mapper {
+class GlobalMapperParams {
+ public:
+  GlobalMapperParams();
+
+  double voxel_xyz0_[3];
+  double voxel_xyz1_[3];
+  double voxel_meters_per_pixel_[3];
+  double voxel_init_value_;
+  double voxel_min_range_;
+  double voxel_max_range_;
+  double voxel_min_z_abs_;
+  double voxel_max_z_abs_;
+  bool voxel_use_rel_cropping_;
+  double voxel_min_z_rel_;
+  double voxel_max_z_rel_;
```

Comment:
We recently had this discussion at the warehouse and I just did another round of reading about it.  The bottom line is that structs can have methods as well and as far as I know, the tendency for structs to have no methods is simply a stylistic holdover with some guarantees on backwards compatibility from C.  As far as I can tell, there is functionally no difference between structs and classes in C++ outside of the fact that structs default to public members and methods.  There is some discussion on stack overflow in the link below about how defining a struct or class as a plain old data (POD) type without any member functions gives you some nice guarantees, but that you lose these if you implement a constructor.  I kinda like having an initializer list.

http://stackoverflow.com/questions/54585/when-should-you-use-a-class-vs-a-struct-in-c